addons/hr_expense/models/hr_expense.py

```diff
@@ -312,3 +304,161 @@ def action_get_attachment_view(self):
+        \nIf the manager approve it, the status is \'Approved\'.\n If the accountant genrate the accounting entries for the expense sheet request, the status is \'Waiting Payment\'.')
+    employee_id = fields.Many2one('hr.employee', string="Employee", required=True, readonly=True, states={'draft': [('readonly', False)]}, default=lambda self: self.env['hr.employee'].search([('user_id', '=', self.env.uid)], limit=1))
+    partner_id = fields.Many2one('res.partner', related="employee_id.address_home_id", string="Employee Home Address")
+    payment_status = fields.Boolean(string='Payment confirmed', help='The employee greens this button when the payment has been received.')
+    payment_mode = fields.Selection([("own_account", "Employee (to reimburse)"), ("company_account", "Company")], compute='_compute_payment_mode', default='own_account', readonly=True, string="Payment By")
+    user_valid = fields.Many2one('res.users', 'Validation By', readonly=True, copy=False, states={'draft': [('readonly', False)], 'submit': [('readonly', False)]})
+    total_amount = fields.Float(string='Total Amount', store=True, compute='_compute_amount', digits=dp.get_precision('Account'))
+    company_id = fields.Many2one('res.company', string='Company', readonly=True, states={'draft': [('readonly', False)]}, default=lambda self: self.env.user.company_id)
+    currency_id = fields.Many2one('res.currency', string='Currency', readonly=True, states={'draft': [('readonly', False)]}, default=lambda self: self.env.user.company_id.currency_id)
+    attachment_number = fields.Integer(compute='_compute_attachment_number', string='Number of Attachments')
+    journal_id = fields.Many2one('account.journal', string='Expense Journal', states={'done': [('readonly', True)], 'post': [('readonly', True)]},
+        default=lambda self: self.env['ir.model.data'].xmlid_to_object('hr_expense.hr_expense_account_journal') or self.env['account.journal'].search([('type', '=', 'purchase')], limit=1),
+        help="The journal used when the expense is done.")
+    accounting_date = fields.Date(default=fields.Date.context_today, string="Accounting Date")
+    bank_journal_id = fields.Many2one('account.journal', string='Bank Journal', states={'done': [('readonly', True)], 'post': [('readonly', True)]}, default=lambda self: self.env['account.journal'].search([('type', 'in', ['case', 'bank'])], limit=1), help="The payment method used when the expense is paid by the company.")
+    account_move_id = fields.Many2one('account.move', string='Journal Entry', copy=False, track_visibility="onchange")
+
+    @api.multi
+    def check_consistency(self):
+        if any(sheet.employee_id != self[0].employee_id for sheet in self):
+            raise UserError(_("Expenses must belong to the same Employee."))
+
+        expense_lines = self.mapped('expense_line_ids')
+        if expense_lines and any(expense.payment_mode != expense_lines[0].payment_mode for expense in expense_lines):
```

Comment:
Same here. Checks if we are reporting several lines paid by the employee AND the companies on the same sheet.